File: playwright_with_ts/pages/login.page.ts
import { Page, expect } from '@playwright/test';

export class LoginPage {
 constructor(private page: Page) {}

 async open() {
   await this.page.goto('https://www.saucedemo.com');
 }

 async login(user: string, pass: string) {

   await expect(this.page.locator('#user-name')).toBeVisible();

   await this.page.locator('#user-name').fill(user);
   await this.page.locator('#password').fill(pass);
   await this.page.locator('#login-button').click();

   await expect(this.page).toHaveURL(/inventory/);
 }
}

File: playwright_with_java/src/test/java/pages/LoginPage.java

package pages;

import com.microsoft.playwright.Page;

import static org.junit.jupiter.api.Assertions.*;

public class LoginPage {

    private final Page page;

    public LoginPage(Page page) {
        this.page = page;
    }

    public void open(String url) {
        page.navigate(url);
        assertTrue(page.locator("#user-name").isVisible());
    }

    public void login(String user, String pass) {
        page.fill("#user-name", user);
        page.fill("#password", pass);
        page.click("#login-button");

        assertTrue(page.url().contains("inventory"));
    }
}
